Modules/modExtendSaveAsText.ACM

```diff
@@ -1126,7 +1134,67 @@ Private Sub AppendParent(AccObj() As AccObject)
   End With
 End Sub
 
-Private Function CleanFormReportFile(strFilename As String, strFilenameNew As String, Ignores() As Ignore) As Boolean
+Public Function CleanFormsReports(strImportPath As String) As Boolean
+  ' cleans out the lines (Sections) specified in the Ignores Array from every Form and Report in the Path Sepcified
+  Dim Ignores() As IGNORE
+  Dim arrSVNItems() As SVNITEM
+  Dim strFoldername As String
+  Dim strFilename As String
+  Dim i As Long
+  Dim j As Long
+  
+  PopIgnoreArr Ignores
+  
+  For i = 0 To 1
+    strFoldername = appendSlash(strImportPath) & IIf(i = 0, "Forms", "Reports")
+      
+    SCC_GetAllVersionsSubFolder strFoldername, arrSVNItems
+    
+    If (Not arrSVNItems) <> -1 Then ' check that the arr is not nothing before looping through it
+      For j = LBound(arrSVNItems) To UBound(arrSVNItems)
+        If arrSVNItems(j).svName <> "" Then
+          strFilename = appendSlash(strFoldername) & arrSVNItems(j).svName
+          If Not CleanFormReportFileEx(strFilename, strFilename, Ignores) Then Exit Function
+        End If
+      Next
+    End If
+  Next
+  
+  CleanFormsReports = True
+
+End Function
+
+Private Function CleanFormReportFileEx(strFilename As String, strFilenameNew As String, Ignores() As IGNORE) As Boolean
+  ' As CleanFormReportFile but accepts any form encoding
+  
+  Dim strEncoding As String
+  
+  CleanFormReportFileEx = False ' intially set to indicate failure
+  
+  If IsUTF8(strFilename) Then
+    strEncoding = "UTF8"
+  ElseIf IsUCS2(strFilename) Then
+    strEncoding = "UCS2"
+  ElseIf IsANSI(strFilename) Then
+    strEncoding = "ANSI"
+  Else
+    Exit Function
+  End If
+  
+  If Not ConvFile_to_UCS2(strFilename, strFilenameNew) Then Exit Function
+  If Not CleanFormReportFile(strFilenameNew, strFilenameNew, Ignores) Then Exit Function
+  
+  Select Case strEncoding
+    Case "UTF8":
+      CleanFormReportFileEx = ConvFileUCS2_to_UTF8(strFilenameNew)
+    Case "UCS2"
+      CleanFormReportFileEx = True
+    Case "ANSI":
+      CleanFormReportFileEx = ConvFileUCS2_to_ANSI(strFilenameNew)
+  End Select
+  
+End Function
+Private Function CleanFormReportFile(strFilename As String, strFilenameNew As String, Ignores() As IGNORE) As Boolean
   ' This function takes a file and cleans out the lines (Sections) specified in the Ignores Array

```

Comment:
nit: meantion that the file has to be UCS2 encoded?

Reply:
Done